### ISSUE TITLE
Can't find profile directory

### ISSUE THREAD

**Issue by Nisthar** [microsoft/playwright#4660]
**Context:**
- GOOD Playwright Version: same version (1.6.2)
- BAD Playwright Version: same version
- Operating System: Windows
- Extra: 

**Code Snippet**

```javascript
process.env.PLAYWRIGHT_BROWSERS_PATH = "0";
process.env.DEBUG = "pw:browser*";
const { firefox } = require("playwright-firefox");
let page, browser;
export async function startAgent() {
  try {
    browser = await firefox.launch({
      headless: false,
      proxy: { server: "per-context" },
    });
 isStarted = true;
  } catch (e) {
    console.log(e);
  }
}
export async function work(proxy = false) {
  try {
    if (!isStarted) {
      await startAgent();
    }
    let context;
   context = await browser.newContext();
  page = await context.newPage();
 }
}
```

**Describe the bug**
This is the logs from `DEBUG=pw:browser*`
```
  [30;1mpw:browser [0m<launching> D:\node_projects\pwtest\node_modules\playwright-firefox\.local-browsers\firefox-1205\firefox\firefox.exe -no-remote -wait-for-browser -foreground -profile C:\Users\mypc\AppData\Local\Temp\playwright_firefoxdev_profile-vrPsOp -juggler-pipe -silent [30m+0ms[0m
  [30;1mpw:browser [0m<launched> pid=23116 [30m+7ms[0m
  [30;1mpw:browser [0m[out]  [30m+1s[0m
  [30;1mpw:browser [0m[out] Juggler listening to the pipe [30m+1ms[0m
  [30;1mpw:browser [0m[err] JavaScript error: resource://gre/modules/XULStore.jsm, line 66: Error: Can't find profile directory. [30m+662ms[0m
  [30;1mpw:browser [0m<gracefully close start> [30m+28s[0m
  [30;1mpw:browser [0m<kill> [30m+4ms[0m
  [30;1mpw:browser [0m<process did exit: exitCode=1, signal=null> [30m+288ms[0m
  [30;1mpw:browser [0m<gracefully close end> [30m+6ms[0m
  [30;1mpw:browser [0m<launching> D:\node_projects\pwtest\node_modules\playwright-firefox\.local-browsers\firefox-1205\firefox\firefox.exe -no-remote -wait-for-browser -foreground -profile C:\Users\mypc\AppData\Local\Temp\playwright_firefoxdev_profile-92fLz0 -juggler-pipe -silent [30m+717ms[0m
  [30;1mpw:browser [0m<launched> pid=7208 [30m+5ms[0m
  [30;1mpw:browser [0m[out]  [30m+1s[0m
  [30;1mpw:browser [0m[out] Juggler listening to the pipe [30m+1ms[0m
  [30;1mpw:browser [0m[err] JavaScript error: resource://gre/modules/XULStore.jsm, line 66: Error: Can't find profile directory. [30m+680ms[0m
  [30;1mpw:browser [0m<gracefully close start> [30m+27s[0m
  [30;1mpw:browser [0m<kill> [30m+3ms[0m
  [30;1mpw:browser [0m<process did exit: exitCode=1, signal=null> [30m+249ms[0m
  [30;1mpw:browser [0m<gracefully close end> [30m+4ms[0m

```

It used to work for me for some time. Had this exact error before. Solved it by reinstalling `playwright-firefox` setting ```
process.env.PLAYWRIGHT_BROWSERS_PATH = "0";
```
it worked for sometime. But its happening again.

**Comment by yury-s** [comment]
The problem is with the profile directory passed to firefox (`C:\Users\mypc\AppData\Local\Temp\playwright_firefoxdev_profile-*`). Looks like it doesn't exist or there are permission issues, it might be that someone deletes it while playwright is running. I'd guess that it might also be free disk space issue on your `C:` drive. Can you check that the directory passed as ` -profile` argument exists while playwright is still running?

**Comment by Nisthar** [comment]
@yury-s I can't find the profile directory in the app data folder. (Sorry if i didn't included that above) I can see a bunch of other `playwright_firefoxdev_profile-*` folder but not the correct one. I have like 95gb free space on the drive. I understand why it works for a few time in the beginning and not now. 

It also worked a few times after i set `process.env.PLAYWRIGHT_BROWSERS_PATH = "0";` and its giving me the same error now.

**Comment by pavelfeldman** [comment]
@yury-s: We auto-create these folders for the execution and auto-delete them after, I don't think they pass any -profile.

**Comment by devmondo** [comment]
i am facing this issue when running multiple instances of firefox in parallel.

running in serial does not cause an issue.

so it might be that each time the next instance starts, it deletes the user profile of the previous one?

if that is the case, shouldn't we create a folder per instance with different name?

thanks!

**Comment by aslushnikov** [comment]
> i am facing this issue when running multiple instances of firefox in parallel.

@devmondo could you please share the snippet to help me reproduce locally?

**Comment by devmondo** [comment]
@aslushnikov thanks for the support!

unfortunately i cant share the private code, but it is nothing special really.


```js
import {firefox} from "playwright-firefox";

async function runBrowser(){
const browser = await firefox.launch(); 

await browser.newContext({
        ignoreHTTPSErrors: true,
        viewport: null,
    });
    }
    
 await Promise.all[runBrowser(),runBrowser()];
    
```

please let me know if i could help more.

thanks!

**Comment by aslushnikov** [comment]
@devmondo ah, two browsers simultaneously! Awesome, I can repro now! We'll fix it shortly
